### ISSUE TITLE
CA1852 doesn't show in Visual Studio; only shows in CLI build

### ISSUE THREAD

**Issue by segevfiner** [dotnet/roslyn-analyzers#7511]
### Analyzer

**Diagnostic ID**: [CA1852](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1852): `Seal internal types`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 9.0.101](https://dotnet.microsoft.com/download/dotnet/9.0)

### Describe the bug

Rule CA1852 shows up on CLI build. but doesn't show in Visual Studi.

### Steps To Reproduce

1. Try this code on a blank new Console App .NET 8/9 SDK project:

```cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    internal class Foo
    {
        private FileStream _bla;

        public Foo()
        {
            _bla = new FileStream(@"c:\temp.txt", FileMode.Open);
        }
    }
}
```

2. On Visual Studio you will see a CA1001 warning but no CA1852.
3. On a CLI build you will see both warnings.

### Expected behavior

You get both warnings in both scenarios.

### Actual behavior

You don't get CA1852 in Visual Studio. Only CA1001.

## Additional context

Windows 11 10.0.22631.0 x64
Visual Studio 2022 17.12.3

![Visual Studio Screenshot](https://github.com/user-attachments/assets/741dceba-9af9-4a9a-ba87-ced8c7650866)
![CLI Screenshot](https://github.com/user-attachments/assets/e99f29ca-b73f-4744-949a-1227686b95d3)

**Comment by marco-carvalho** [comment]
I have a project with at least +1000 warnings of CA1852, but I can't auto fix it (like I can with CA1805, CA1825, CA1872). There is a way to apply a autofix to resolve all CA1852 in a solution?

**Comment by koryukov** [comment]
There is the same issue with Visual Studio 17.13.6.

**Comment by ryanelian** [comment]
I can confirm this is an also issue in Visual Studio Code + .NET 10

**Comment by ryanelian** [comment]
Solved this by adding the following settings to my `.vscode/settings.json`

```json
{
	"dotnet.backgroundAnalysis.analyzerDiagnosticsScope": "fullSolution",
	"dotnet.backgroundAnalysis.compilerDiagnosticsScope": "fullSolution"
}
```

**Comment by DemoBytom** [comment]
I'm hitting the same issue in Visual Studio 2026 version 18.3.0 with .NET 10.0.102.
I've set .enditorconfig:
```
dotnet_diagnostic.CA1852.severity = warning
dotnet_code_quality.CA1852.ignore_internalsvisibleto = true
```
And I'm using `WarningsAsError`. The compilation breaks, correct info is in the build Output, but the errors are not reported in the Error List, nor in code.

A solution/workaround seems to be the same as @ryanelian's : `options/C#/Advanced -> Run background code analysis for: Entire solution`. By default it was set to `Current document`:

<img width="1083" height="613" alt="Image" src="https://github.com/user-attachments/assets/5006a5c5-1eb8-41c7-988d-c59a57967f67" />

This caused the errors to reappear in the code and in the Error List
